spec/v3/spec.md

```diff
@@ -1,93 +1,372 @@
+
+### 3.1 Overview
+```
+Offset     00   01   02   03   04   05   06   07   08   09   0A   0B   0C   0D   0E   0F
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000000   |           Magic Number           |  V |         Root Directory Offset         |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000010   |         Root Directory Length         |            Metadata Offset            |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000020   |            Metadata Length            |        Leaf Directories Offset        |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000030   |        Leaf Directories Length        |            Tile Data Offset           |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000040   |            Tile Data Length           |         Num of Addressed Tiles        |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000050   |         Number of Tile Entries        |        Number of Tile Contents        |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000060   |  C | IC | TC | TT |MinZ|MaxZ|              Min Position             |      Max 
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000070    Position                     |CenZ|            Center Position            |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+```
+
+### 3.2 Fields
+#### Magic Number
+
+The magic number is a fixed 7-byte field whose value is always `PMTiles` in UTF-8 encoding (`0x50 0x4D 0x54 0x69 0x6C 0x65 0x73`)
+
+#### Version (V)
+
+The version is a fixed 1-byte field whose value is always 3 (`0x03`).
+
+#### Root Directory Offset
+
+The Root Directory Offset is a 8-byte field whose value gives the offset of the first byte of the root directory. This address offset is relative to the first byte of the archive.
+
+This field is encoded as an little-endian 64-bit unsigned integer.
+
+#### Root Directory Length
+
+The Root Directory Length is a 8-byte field specifying the number of bytes in the root directory.
+
+This field is encoded as an little-endian 64-bit unsigned integer.
+
+#### Metadata Offset
+
+The Metadata Offset is a 8-byte field whose value gives the offset of the first byte of the metadata. This address offset is relative to the first byte of the archive.
+
+This field is encoded as an little-endian 64-bit unsigned integer.
+
+#### Metadata Length
+
+The Metadata Length is a 8-byte field specifying the number of bytes reserved for the metadata. A value `0` indicates that there is no metadata included in this PMTiles archive.
```

Comment:
@bdon in https://github.com/protomaps/PMTiles/issues/107#issuecomment-1535056512:

> I think this could better be "number of bytes used", since reserved might imply there is empty space in a typical file.

Reply:
Theoretically you could reserve more space for any section (apart from the meta data) than it actually needs 🤔. Or am I getting something wrong here? 🤔 For example: I could reserve 30MB for the tile data section, but only have 10MB of tile data.

Reply:
Continued [here](https://github.com/protomaps/PMTiles/pull/172#discussion_r1266232148)